android/src/test/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapperTests.kt

```diff
@@ -0,0 +1,129 @@
+package com.revenuecat.purchases.hybridcommon.mappers
+
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.models.StoreProduct
+import org.assertj.core.api.Assertions.assertThat
+import org.json.JSONObject
+import org.junit.jupiter.api.Test
+
+internal class StoreProductMapperTest {
+
+    @Test
+    fun `maps product identifier correctly`() {
+        val expected = "expected_product_identifier"
+        stubStoreProduct(sku = expected).map().let {
+            assertThat(it["identifier"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps product description correctly`() {
+        val expected = "Expected Description"
+        stubStoreProduct(description = expected).map().let {
+            assertThat(it["description"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps product title correctly`() {
+        val expected = "Expected Title"
+        stubStoreProduct(title = expected).map().let {
+            assertThat(it["title"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps product price correctly`() {
+        val expected = 2.0
+        stubStoreProduct(priceAmountMicros = (expected * 1_000_000).toLong()).map().let {
+            assertThat(it["price"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps product price string correctly`() {
+        val expected = "$2.00"
+        stubStoreProduct(price = expected).map().let {
+            assertThat(it["priceString"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps currency code correctly`() {
+        val expected = "CAD"
+        stubStoreProduct(priceCurrencyCode = expected).map().let {
+            assertThat(it["currencyCode"]).isEqualTo(expected)
+        }
+    }
+
+    @Test
+    fun `maps discounts correctly`() {
+        stubStoreProduct().map().let {
+            assertThat(it["discounts"]).isEqualTo(null)
```

Comment:
Hmm I don't think we have any tests where this is not null?

Reply:
oh true, good catch

Reply:
oh, actually, it's always null in Android lol